internal/plugin/collector.go

```diff
@@ -86,28 +100,39 @@ func (c *Collector) GetAllEnabledAndUsedPlugins(cfg *config.Config) ([]string, [
 
 			_, found := bindExecutors[groupName]
 			if !found {
-				l.Debug("Executor plugin defined and enabled but not used by any platform")
+				l.Debug("Executor plugin defined and enabled but not used by any platform or standalone action")
 				continue
 			}
 
-			usedExecutorPlugins = append(usedExecutorPlugins, name)
+			l.Debug("Marking executor plugin as enabled")
+			usedExecutorPlugins[name] = struct{}{}
 		}
 	}
 
 	// Collect all sources that are both enabled and bind to at least one communicator that is enabled.
-	var usedSourcePlugins []string
+	usedSourcePlugins := map[string]struct{}{}
 	for groupName, groupItems := range cfg.Sources {
 		for name, source := range groupItems.Plugins {
+			l := c.log.WithFields(logrus.Fields{
+				"groupName": groupName,
+				"pluginKey": name,
+			})
+
 			if !source.Enabled {
+				l.Debug("Source plugin defined but not enabled.")
+
 				continue
 			}
 			_, found := bindSources[groupName]
 			if !found {
+				l.Debug("Source plugin defined and enabled but not used by any platform or standalone action")
 				continue
 			}
 
-			usedSourcePlugins = append(usedSourcePlugins, name)
+			l.Debug("Marking source plugin as enabled")
+			usedSourcePlugins[name] = struct{}{}
 		}
 	}
-	return usedExecutorPlugins, usedSourcePlugins
+
+	return maps.Keys(usedExecutorPlugins), maps.Keys(usedSourcePlugins)
```

Comment:
Can you also use `maps.Keys` in `maputil.SortKeys`? Thanks!

I also searched for `sort.Strings` and saw other places where we can use it - e.g. in kubectl_cmd_builder.go:

```go
	for key := range enabledKubectls.AllowedKubectlResource {
		resources = append(resources, key)
	}
	sort.Strings(resources)

	for key := range enabledKubectls.AllowedKubectlVerb {
		verbs = append(verbs, key)
	}
	sort.Strings(verbs)
```

Reply:
the `kubectl_cmd_builder.go` will be removed soon so I just skipped but I can change it to 👍

Reply:
Skip it then 👍 I forgot that this is the legacy builder.